### ISSUE TITLE
Webdriver.executeScript/Webdriver.schedule 'Method is not implemented' Error

### ISSUE THREAD

**Issue by flashquartermaster** [angular/protractor#3616]
- Node Version: `v4.4.5`
- Protractor Version: `4.0.9`
- Angular Version: `1.5.3`
- Browser(s): `N/a`
- Operating System and Version `macOS Sierra 10.12`

- Your protractor configuration file:
```
var appLocation = 'Absolute/path/to/the.app';
exports.config = {
    seleniumAddress: 'http://localhost:4723/wd/hub',
    specs: [
        'features/*.feature'
    ],
    framework: 'custom',
    frameworkPath: require.resolve('protractor-cucumber-framework'),
    cucumberOpts: {
        require: ['features/step_definitions/*_steps.js', 'features/support/*.js'],
        format: 'pretty'
    },
    capabilities: {
        'appium-version': '1.6.0-beta2',
        automationName: 'XCUITest',
        platformName: 'iOS',
        platformVersion: '10.0',
        deviceName: 'iPhone 6s Plus',
        browserName: "",
        autoWebview: true,
        fullReset: true,
        app: appLocation
    },
    baseUrl: '',
    onPrepare: function () {
        var wd = require('wd'),
            protractor = require('protractor'),
            wdBridge = require('wd-bridge')(protractor, wd);

        wdBridge.initFromProtractor(exports.config);

        //To navigate using file:// rather than http://
        var defer = protractor.promise.defer();
        browser.ignoreSynchronization = true;
        browser.executeScript( 'return window.location;' ).then( function(location){
            browser.resetUrl = 'file://';
            browser.baseUrl = location.origin + location.pathname;
            defer.fulfill();
        });
        return defer.promise;
    }
};
```

- A relevant example test:
https://github.com/flashquartermaster/myappname with capabilities upgarded to reflect:
'appium-version': '1.6.0-beta2',
automationName: 'XCUITest',

- Output from running the test
```
$ ../node_modules/.bin/protractor ios-conf.js --troubleshoot
[17:19:14] D/launcher - Running with --troubleshoot
[17:19:14] D/launcher - Protractor version: 4.0.9
[17:19:14] D/launcher - Your base url for tests is 
[17:19:14] I/hosted - Using the selenium server at http://localhost:4723/wd/hub
[17:19:14] I/launcher - Running 1 instances of WebDriver
[17:20:48] D/runner - WebDriver session successfully started with capabilities Capabilities {
  'webStorageEnabled' => false,
  'locationContextEnabled' => false,
  'browserName' => '',
  'platform' => 'MAC',
  'javascriptEnabled' => true,
  'databaseEnabled' => false,
  'takesScreenshot' => true,
  'networkConnectionEnabled' => false,
  'appium-version' => '1.6.0-beta2',
  'automationName' => 'XCUITest',
  'platformName' => 'iOS',
  'platformVersion' => '10.0',
  'deviceName' => 'iPhone 6s Plus',
  'autoWebview' => true,
  'fullReset' => true,
  'app' => '/Absolute/path/to/myappname.app',
  'count' => 1 }
[17:20:48] D/runner - Running with spec files /PathTo/myappname/tests/features/login.feature
[17:20:50] E/launcher - Method is not implemented
[17:20:50] E/launcher - WebDriverError: Method is not implemented
    at WebDriverError (/PathTo/myappname/node_modules/selenium-webdriver/lib/error.js:27:10)
    at Object.checkLegacyResponse (/PathTo/myappname/node_modules/selenium-webdriver/lib/error.js:639:15)
    at parseHttpResponse (/PathTo/myappname/node_modules/selenium-webdriver/http/index.js:538:13)
    at /PathTo/myappname/node_modules/selenium-webdriver/http/index.js:472:11
    at ManagedPromise.invokeCallback_ (/PathTo/myappname/node_modules/selenium-webdriver/lib/promise.js:1379:14)
    at TaskQueue.execute_ (/PathTo/myappname/node_modules/selenium-webdriver/lib/promise.js:2913:14)
    at TaskQueue.executeNext_ (/PathTo/myappname/node_modules/selenium-webdriver/lib/promise.js:2896:21)
    at /UPathTo/myappname/node_modules/selenium-webdriver/lib/promise.js:2820:25
    at /PathTo/myappname/node_modules/selenium-webdriver/lib/promise.js:639:7
    at process._tickCallback (node.js:369:9)
From: Task: WebDriver.executeScript()
    at WebDriver.schedule (/PathTo/myappname/node_modules/selenium-webdriver/lib/webdriver.js:377:17)
    at WebDriver.executeScript (/PathTo/myappname/node_modules/selenium-webdriver/lib/webdriver.js:526:16)
    at ProtractorBrowser.to.(anonymous function) [as executeScript] (/PathTo/myappname/node_modules/protractor/built/browser.js:62:29)
    at exports.config.onPrepare (/PathTo/myappname/tests/ios-conf.js:47:17) (EDIT: Refers to this line: browser.executeScript('return window.location;').then( function(location){)
    at /PathTo/myappname/node_modules/protractor/built/util.js:48:49
    at Function.promise (/PathTo/myappname/node_modules/q/q.js:682:9)
    at Object.runFilenameOrFn_ (/PathTo/myappname/node_modules/protractor/built/util.js:38:16)
    at /PathTo/myappname/node_modules/protractor/built/runner.js:93:27
    at _fulfilled (/PathTo/myappname/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/PathTo/myappname/node_modules/q/q.js:863:30)
[17:20:50] E/launcher - Process exited with error code 199
```

- Steps to reproduce the bug

Run the example above
- The URL you are running your tests against (if relevant)

N/a

**Comment by flashquartermaster** [comment]
@mgiambalvo I'm wondering if appium is the problem. I just upgraded to 1.6 beta 3 from beta 2 and I have a different error now
```
[17:21:26] E/launcher - An unknown server-side error occurred while processing the command. Original error: Unexpected token C
[17:21:26] E/launcher - WebDriverError: An unknown server-side error occurred while processing the command. Original error: Unexpected token C
    at WebDriverError (/PathTo/myappname/node_modules/selenium-webdriver/lib/error.js:27:10)
    at Object.checkLegacyResponse (/PathTo/myappname/node_modules/selenium-webdriver/lib/error.js:639:15)
    at parseHttpResponse (/PathTo/myappname/node_modules/selenium-webdriver/http/index.js:538:13)
    at /PathTo/myappname/node_modules/selenium-webdriver/http/index.js:472:11
    at ManagedPromise.invokeCallback_ (/PathTo/myappname/node_modules/selenium-webdriver/lib/promise.js:1379:14)
    at TaskQueue.execute_ (/PathTo/myappname/node_modules/selenium-webdriver/lib/promise.js:2913:14)
    at TaskQueue.executeNext_ (/PathTo/myappname/node_modules/selenium-webdriver/lib/promise.js:2896:21)
    at /PathTo/myappname/node_modules/selenium-webdriver/lib/promise.js:2820:25
    at /PathTo/myappname/node_modules/selenium-webdriver/lib/promise.js:639:7
    at process._tickCallback (node.js:369:9)
From: Task: WebDriver.createSession()
    at Function.createSession (/PathTo/myappname/node_modules/selenium-webdriver/lib/webdriver.js:329:24)
    at Builder.build (/PathTo/myappname/node_modules/selenium-webdriver/builder.js:458:24)
    at Hosted.DriverProvider.getNewDriver (/PathTo/myappname/node_modules/protractor/built/driverProviders/driverProvider.js:37:33)
    at Runner.createBrowser (/PathTo/myappname/node_modules/protractor/built/runner.js:198:43)
    at /PathTo/myappname/node_modules/protractor/built/runner.js:277:30
    at _fulfilled (/PathTo/myappname/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/PathTo/myappname/node_modules/q/q.js:863:30)
    at Promise.promise.promiseDispatch (/PathTo/myappname/node_modules/q/q.js:796:13)
    at /PathTo/myappname/node_modules/q/q.js:556:49
    at runSingle (/PathTo/myappname/node_modules/q/q.js:137:13)
[17:21:26] E/launcher - Process exited with error code 199
```

**Comment by mgiambalvo** [comment]
Thanks for following up! I also suspect Appium, however I've had trouble setting up a test environment.

Could you try Appium 1.5.3 and let me know how it goes? Thanks.

**Comment by flashquartermaster** [comment]
Hi,

This issue has arisen since I upgraded to Xcode 8 and had to move from Appium 1.5.3 to 1.6.0-beta to be able to use 'XCUITest' instead of 'UIAutomation'.

However, I have got to the bottom of it, it is a conflict with `ionic serve`. I think it is because Appium and Ionic are both using localhost:8100. Basically, I am getting a bad response when creating a new session.

Ionic:
`√ Running dev server:  http://localhost:8100`

Appium:
`[JSONWP Proxy] Proxying [POST /session] to [POST http://localhost:8100/session]`

Everything is running ok now and the Webdriver.executeScript/Webdriver.schedule 'Method is not implemented' Error has gone away, to the extent that I cannot reproduce it with Appium 1.6.0-beta2.

Should I log something with Appium?
